mailbox/maildir/src/main/java/org/apache/james/mailbox/maildir/mail/MaildirMailboxMapper.java

```diff
@@ -267,9 +272,15 @@ public void endRequest() {
      * @return The id of the cached mailbox
      */
     private Mailbox cacheMailbox(Mailbox mailbox) {
-        mailboxCache.add(new SimpleMailbox(mailbox));
-        int id = mailboxCache.size() - 1;
-        ((SimpleMailbox) mailbox).setMailboxId(MaildirId.of(id));
+        boolean isCached = mailboxCache.keySet().contains(mailbox);
```

Comment:
given two same `mailbox1`, `mailbox2` but no mailbox have id. Consider following statement calls:
```java
cacheMailbox(mailbox1); // will cache mailbox1 with new id
cacheMailbox(mailbox2); // isCached will be false because mailbox1.equals(mailbox2) = false (different id), then it will cache mailbox2 also, this is causing redundant data from the cache. Do you think?
```

Reply:
>  this is causing redundant data from the cache. Do you think?

My commit address this